Classes/DataProcessing/ContainerProcessor.php

```diff
@@ -103,18 +102,15 @@ protected function processColPos(
     ): array {
         $children = $container->getChildrenByColPos($colPos);
 
-        $typo3Version = GeneralUtility::makeInstance(Typo3Version::class);
-        if ($typo3Version->getMajorVersion() < 12) {
-            $contentRecordRenderer = new RecordsContentObject($cObj);
-        } else {
-            $contentRecordRenderer = new RecordsContentObject();
-            $contentRecordRenderer->setContentObjectRenderer($cObj);
-            $contentRecordRenderer->setRequest($this->getRequest());
+        $contentRecordRenderer = $cObj->getContentObject('RECORDS');
+        if ($contentRecordRenderer === null) {
```

Comment:
@ervaude when can this occure? (cannot imaging?) i would prefer to remove this check, and the Exception?

Reply:
The check is only "useful" for phpstan because `$cObj->getContentObject('RECORDS');` could return `null`.

But if the `RECORDS` cObject is not registered TYPO3 would not function at all.

When we remove the check we'll need a phpstan ignore-error entry.